src/client/power-pages/metadata-diff/MetadataDiffCommands.ts

```diff
@@ -0,0 +1,615 @@
+            if (fs.existsSync(storagePath)) {
+                fs.rmSync(storagePath, { recursive: true, force: true });
+            }
+            fs.mkdirSync(storagePath, { recursive: true });
+
+            const progressOptions: vscode.ProgressOptions = {
+                location: vscode.ProgressLocation.Notification,
+                title: vscode.l10n.t("Downloading website metadata"),
+                cancellable: false
+            };
+
+            let pacPagesDownload;
+            let comparisonBuilt = false;
+            await vscode.window.withProgress(progressOptions, async (progress) => {
+                progress.report({ message: "Looking for this website in the connected environment..." });
+                const pacPagesList = await MetadataDiffDesktop.getPagesList(pacTerminal);
+                if (pacPagesList && pacPagesList.length > 0) {
+                    const websiteRecord = pacPagesList.find((record) => record.id === websiteId);
+                    if (!websiteRecord) {
+                        vscode.window.showErrorMessage("Website not found in the connected environment.");
+                        return;
+                    }
+                    progress.report({ message: vscode.l10n.t('Retrieving "{0}" as {1} data model. Please wait...', websiteRecord.name, websiteRecord.modelVersion === 'v2' ? 'enhanced' : 'standard') });
+                    pacPagesDownload = await pacWrapper.pagesDownload(
+                        storagePath,
+                        websiteId,
+                        websiteRecord.modelVersion === "v1" || websiteRecord.modelVersion === "Standard" ? "1" : "2"
+                    );
+                    if (pacPagesDownload) {
+                        progress.report({ message: vscode.l10n.t('Comparing metadata of "{0}"...', websiteRecord.name) });
+                        const provider = MetadataDiffTreeDataProvider.initialize(context);
+                        MetadataDiffDesktop.setTreeDataProvider(provider);
+                        ActionsHubTreeDataProvider.setMetadataDiffProvider(provider);
+                        await provider.getChildren();
+                        vscode.commands.executeCommand("microsoft.powerplatform.pages.actionsHub.refresh");
+                        comparisonBuilt = true;
+                    }
+                }
+            });
+
+            if (pacPagesDownload && comparisonBuilt) {
+                vscode.window.showInformationMessage(vscode.l10n.t("You can now view the comparison"));
+            } else {
+                vscode.window.showErrorMessage("Failed to download metadata.");
+            }
+
+        } catch (error) {
+            oneDSLoggerWrapper.getLogger().traceError(Constants.EventNames.METADATA_DIFF_REFRESH_FAILED, error as string, error as Error, { methodName: null }, {});
+        }
+    });
+
+    vscode.commands.registerCommand("microsoft.powerplatform.pages.metadataDiff.triggerFlow", async () => {
```

Comment:
Can we please refactor this file?
1. When registering commands, you can register single methods instead of defining the logic inline.
2. The logic itself can be broken into multiple reusable functions. A single function seems to be doing too much right now.